build/Jenkinsfile.pr

```diff
@@ -125,14 +125,22 @@ pipeline {
                 withDockerContainer(args: "-u root --privileged --net ${COMPOSE_PROJECT_NAME}_mm-test -v ${WORKSPACE}/src:/go/src/", image: 'mattermost/mattermost-build-server:dec-7-2018') {
                     ansiColor('xterm') {
                     sh """
-                            cd /go/src/github.com/mattermost/mattermost-server
+                        cd /go/src/github.com/mattermost/mattermost-server
+
+                        mkdir -p client/plugins
+                        cat config/config.json
 
-                            mkdir -p client/plugins
-                            cat config/config.json
+                        make test-server BUILD_NUMBER='${BRANCH_NAME}-${BUILD_NUMBER}' TESTFLAGS= TESTFLAGSEE=
 
-                            make test-server BUILD_NUMBER='${BRANCH_NAME}-${BUILD_NUMBER}' TESTFLAGS= TESTFLAGSEE=
                         """
                     }
+                    withCredentials([string(credentialsId: 'CODECOV_TOKEN', variable: 'CODECOV')]) {
+                        sh '''#!/bin/bash
+                            set -x
+                            cd /go/src/github.com/mattermost/mattermost-server
+                            bash <(curl -s https://codecov.io/bash) -t $CODECOV
```

Comment:
Is it feasible to host https://codecov.io/bash ourselves? This is not a blocking request, given the large number of things we download dynamically to build our code, but if straightforward, it would be lovely to plug an attack vector should codecov.io be compromised.

Reply:
In the codecov docs, this is the only option we have to upload the report. Off course we can download the bash script and avoid the curl every time, however, if they update the script to a newer version we will need to rebuild the Jenkins image to update that. I prefer to keep as is 3/5.